src/LinAlg/matrix.h

```diff
@@ -1367,11 +1370,13 @@ namespace utl //! General utility classes and functions.
 
   template<class T> inline
   vector<T>& vector<T>::add(const std::vector<T>& X, const T& alfa,
-                            int ofsx, int stridex)
+                            int ofsx, int stridex,
+                            int ofsy, int stridey)
   {
-    T* p = this->ptr();
+    T* p = this->ptr() + ofsy;
     const T* q = X.data() + ofsx;
-    for (size_t i = 0; i < this->size() && i < X.size(); i++, p++, q += stridex)
+    size_t size = std::min(this->size() / stridey, X.size() / stridex);
+    for (size_t i = 0; i < size; ++i, p += stridey, q += stridex)
```

Comment:
If either ofsx or ofsy are non-zero (positive), I think the loop bounds will be wrong here?

Reply:
there's a theoretical issue yes, but not with sane input data. the default blas implementations will crash in the same way so i think i'll just keep the assumptions on input data.

Reply:
ok, since this is only for the (never used) no-blas fallback anyway